includes/wcs-renewal-functions.php

```diff
@@ -142,6 +149,30 @@ function wcs_get_last_non_early_renewal_order( $subscription ) {
 	return $last_non_early_renewal;
 }
 
+/**
+ * Get the last renewal order (early renewals included).
+ *
+ * @param WC_Subscription $subscription The subscription object.
+ * @return WC_Order|bool The last non-early renewal order, otherwise false.
+ */
+function wcs_get_last_renewal_order( $subscription ) {
+	$renewal_orders = wcs_get_newest_renewal_orders( $subscription );
+	return $renewal_orders ? reset( $renewal_orders ) : false;
+}
+
+/**
+ * @param $subscription WC_Subscription The subscription object.
+ * @return array The latest renewal orders
+ */
+function wcs_get_newest_renewal_orders( $subscription ) {
+	$renewal_orders = $subscription->get_related_orders( 'all', 'renewal' );
+
+	// We need the orders sorted by the date they were created, with the newest first.
+	wcs_sort_objects( $renewal_orders, 'date_created', 'descending' );
+
+	return $renewal_orders;
+}
```

Comment:
I think this function maybe too narrow in scope. It serves the purpose in this function by returning the orders in the descending date created order but I think it would be better to be more general with something like this:


```php
/**
 * Gets the renewal orders for a subscription, sorted by the specified property.
 *
 * @param WC_Subscription $subscription The subscription object.
 * @param string          $sort_by      The subscription property to sort by.
 * @param string          $order        Optional. The sort order to sort by. Default is 'descending'.
 *
 * @return WC_Order[] The subscriptions renewal orders sorted.
 */
function wcs_get_renewal_orders_sorted_by( $subscription, $sort_by, $order = 'descending' ) {
	$renewal_orders = $subscription->get_related_orders( 'all', 'renewal' );

	// We need the orders sorted by the date they were created, with the newest first.
	wcs_sort_objects( $renewal_orders, $sort_by, $order );

	return $renewal_orders;
}
```

Reply:
Good idea! This will make the function more useful. I updated my code following your suggestion 👍